app/src/main/java/template/MainActivity.kt

```diff
@@ -3,36 +3,72 @@ package template
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.systemBarsPadding
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Surface
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
-import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.core.view.WindowCompat
+import com.google.accompanist.systemuicontroller.rememberSystemUiController
 import template.theme.TemplateTheme
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        configureEdgeToEdgeWindow()
+
         setContent {
+            ConfigureTransparentSystemBars()
+
             TemplateTheme {
-                // A surface container using the 'background' color from the theme
-                Surface(color = MaterialTheme.colorScheme.background) {
+                Surface(
+                    color = MaterialTheme.colorScheme.background,
+                    modifier = Modifier
+                        .systemBarsPadding(),
```

Comment:
Hmm, while this is definitely a sane default for the case where you want to simply have transparent system bars, it's not if you want your content to go behind them when it's able to (when scrolling for example). With this, since the padding is in the entire screen itself, the content would get cut-off as it's scrolling away.

Do you feel like the template should lean towards being cautious, and have this here and for those who know that they want to do that they can remove this so that it's easier for people not familiar with the inset APIs, or should the template not make such an assumption and let all insets handling happen from the user themselves, whenever they actually need to.

Either way works fine for me, I just know that if I copied from this template, this is the first thing I'd remove.

Reply:
Great feedback! Initially, I was worried _not_ to include it, because then the default AS template looks funny. That said, since the "template" UI is just this basic surface, it's probably better not to include this so I don't confuse folks, and let people who clone it worry about the appropriate content padding behaviors. I don't wanna be to opinionated about how the outer surface container should work.

Reply:
Yeap, I understand your initial thoughts. Not having an opinionated container probably fits best in such a template after all. Thanks for considering my feedback 🤗